engineapi/instance_manager.go

```diff
@@ -133,16 +157,24 @@ func NewInstanceManagerClient(im *longhorn.InstanceManager) (*InstanceManagerCli
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to load Instance Manager Instance Service Client TLS files")
 		}
-		if _, err = instanceClient.VersionGet(); err != nil {
-			return nil, errors.Wrap(err, "failed to check version of Instance Manager Instance Service Client with TLS connection")
+
+		if err = instanceServiceClient.CheckConnection(); err != nil {
+			return instanceServiceClient, errors.Wrapf(err, "failed to check Instance Manager Instance Service Client TLS connection for %v IP %v",
+				im.Name, im.Status.IP)
 		}
 
-		return instanceClient, nil
+		if _, err = instanceServiceClient.VersionGet(); err != nil {
+			return instanceServiceClient, errors.Wrap(err, "failed to check version of Instance Manager Instance Service Client with TLS connection")
+		}
+
+		return instanceServiceClient, nil
 	}
 
 	initDiskServiceTLSClient := func(endpoint string) (*imclient.DiskServiceClient, error) {
+		ctx, cancel := context.WithCancel(context.Background())
+
 		// check for tls cert file presence
-		diskClient, err := imclient.NewDiskServiceClientWithTLS(endpoint,
+		diskClient, err := imclient.NewDiskServiceClientWithTLS(ctx, cancel, endpoint,
```

Comment:
Won't we have a similar defer closure to clean up `diskClient` if encountering an error?

Reply:
Pushed the codes to wrong repo.
`NewDiskServiceClientWithTLS` is not used, because it is not supported yet.
Please review again. Thank you.